lib/hdb/common.c

```diff
@@ -1307,6 +1306,138 @@ make_namespace_princ(krb5_context context,
     return ret;
 }
 
+static int
+is_namespace_princ_p(krb5_context context,
+                     krb5_const_principal princ)
+{
+    return
+        krb5_principal_get_num_comp(context, princ) >= 4
+        && strcmp(krb5_principal_get_comp_string(context, princ, 0),
+                  "WELLKNOWN") == 0
```

Comment:
If used in more than one place, perhaps a macro in `krb5.h` would be better.

Reply:
I mean, it will never change...